### PR TITLE
Adds private address for uphold cards

### PR DESCRIPTION
**Pull request by NejcZdovc**
Resolves https://github.com/brave/brave-browser/issues/5547

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [ ] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [ ] Ran `git rebase master` (if needed).
- [ ] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:
- enable rewards
- connect to KYC uphold account
 - got to Preference file
 - make sure that there is `uphold_anon_address` and has address in there
 - write address somewhere
 - close browser
- clean profile
- enable rewards
- connect to KYC uphold account (same as on step 2)
 - got to Preference file
 - make sure that there is `uphold_anon_address` and has address that is the same as it was on step 4

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_card.h [hunk 1]
```diff
@@ -76,6 +84,40 @@ class UpholdCard {
...
+      std::map<std::string, std::string> addresses,
+      const ledger::ExternalWallet& wallet,
+      CreateAnonAddressCallback callback);
+
+  void CreateAnonAddress(
+      ledger::ExternalWalletPtr wallet,
+      CreateAnonAddressCallback callback);
+
+  void OnCreateAnonAddress(
+      int response_status_code,
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_card.h]: nit: const

File: vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_card.h [hunk 2]
```diff
@@ -76,6 +84,40 @@ class UpholdCard {
...
+      GetCardAddressesCallback callback);
+
+  void OnCreateAnonAddressGet(
+      ledger::Result result,
+      std::map<std::string, std::string> addresses,
+      const ledger::ExternalWallet& wallet);
+
+  void OnCreateAnonAddressIfNecessary(
+      ledger::Result result,
+      std::map<std::string, std::string> addresses,
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_card.h]: `const std::map<std::string, std::string>& addresses`

File: vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_card.h [hunk 3]
```diff
@@ -76,6 +84,40 @@ class UpholdCard {
...
+
+  void OnGetCardAddresses(
+      int response_status_code,
+      const std::string& response,
+      const std::map<std::string, std::string>& headers,
+      GetCardAddressesCallback callback);
+
+  void OnCreateAnonAddressGet(
+      ledger::Result result,
+      std::map<std::string, std::string> addresses,
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_card.h]: `const std::map<std::string, std::string>& addresses`

File: vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_card.h [hunk 4]
```diff
@@ -76,6 +84,40 @@ class UpholdCard {
...
 
+  void GetCardAddresses(
+      ledger::ExternalWalletPtr wallet,
+      GetCardAddressesCallback callback);
+
+  std::map<std::string, std::string> ParseGetCardAddressResponse(
+      const std::string& response);
+
+  void OnGetCardAddresses(
+      int response_status_code,
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/uphold/uphold_card.h]: `const int`

File: vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.cc [hunk 5]
```diff
@@ -263,31 +266,34 @@ void Wallet::OnTransferAnonToExternalWallet(
...
                _1,
                _2,
-               new_address,
+               *wallet,
                allow_zero_balance,
                callback));
 }
 
 void Wallet::OnTransferAnonToExternalWalletBalance(
     ledger::Result result,
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.cc]: const

File: vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.h [hunk 6]
```diff
@@ -104,11 +104,23 @@ class Wallet {
     ledger::TransferAnonToExternalWalletCallback callback);
 
   void OnTransferAnonToExternalWalletBalance(
-    ledger::Result result,
-    ledger::BalancePtr properties,
-    const std::string& new_address,
-    const bool allow_zero_balance,
-    ledger::TransferAnonToExternalWalletCallback callback);
+      ledger::Result result,
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.h]: const

File: vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.h [hunk 7]
```diff
@@ -104,11 +104,23 @@ class Wallet {
...
+      const bool allow_zero_balance,
+      ledger::TransferAnonToExternalWalletCallback callback);
+
+  std::string GetClaimPayload(
+      const std::string user_funds,
+      const std::string new_address,
+      const std::string anon_address);
+
+  void OnTransferAnonToExternalWalletAddress(
+      ledger::Result result,
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.h]: const

File: vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.cc [hunk 8]
```diff
@@ -339,31 +361,95 @@ void Wallet::OnTransferAnonToExternalWalletBalance(
...
-                                                             header_values,
-                                                             1,
-                                                             "primary",
-                                                             new_secret_key);
+  if (result != ledger::Result::ALREADY_EXISTS) {
+    ledger_->SetStringState(ledger::kStateUpholdAnonAddress, anon_address);
+  }
+
+  const std::string path = base::StringPrintf(
+      "%s%s/claim",
```
Comment (tmancey) [inline vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.cc]: %s%s is this correct or should it be %s/%s

Comment (NejcZdovc) [reply vendor/bat-native-ledger/src/bat/ledger/internal/wallet/wallet.cc]: this is correct

**Pull request merged**